### ISSUE TITLE
Error

### ISSUE THREAD

**Issue by CodersBrothers** [z0mt3c/node-restify-validation#26]
in 0.1.3 returns this when I want validate some param that not exists and is requried:
```json
{
"code": "InternalError",
"message": "\"string\" == \"object\""
}
```

I think that error.js in lib folder not it's correct because if I add the 0.1.0 javascript code of error.js in the new version this not return error, but not return information too.

**Comment by gchauvet** [comment]
- Seems specific in 0.1.3.
- No issue on 1.0.3

Can you write a small unit case plz ?

**Comment by CodersBrothers** [comment]
It's really weird. In npm the version is 0.1.3, here the last release is 0.1.2

I don't understand this of "1.0.3"

**Comment by gchauvet** [comment]
Yes, this is curious, because node-restify-validation is tagged as "0.1.3" on npm repository, but github package.json is set to "1.0.3".

In despit of this, I've updated my project with the npm registred version, but I can't reproduce this bug with my own uses cases.

**Comment by CodersBrothers** [comment]
Maybe I use as wrong way. I can resume as:

I add this when I create server:
```js
server.use(restifyValidation.validationPlugin({
        errorsAsArray: true,
        errorHandler: restify.errors.InvalidArgumentError
    }));
```

And when I and set validate params:
```js
server.get({
            url: route.url,
            validation: {
            "country": {
                "description": "Country code (ISO3166). Example: MX, ES.",
                "scope": "body",
                "isRequired": true
            }
            }
        }, function(req, res, next){
});
```

**Comment by gchauvet** [comment]
okay, the problem is due to a major change with scope (resources/queries/content = body)
the new way in "1.0.3" (but released as "0.1.3" on npm repository) is : 

```javascript
server.get({
            url: route.url,
            validation: {
                queries : {
                    country: {
                        "description": "Country code (ISO3166). Example: MX, ES.",
                        "scope": "body",
                        "isRequired": true
                   }
                }
            }
        }, function(req, res, next){
});
```

See releaded documentation on node-restify-validation project page

**Comment by CodersBrothers** [comment]
Let me some minutes to check and test. Thanks.

**Comment by gchauvet** [comment]
#25

**Comment by CodersBrothers** [comment]
The same error, was imposible run in 0.1.3

**Comment by z0mt3c** [comment]
Sorry for mixing things up... We now have version 0.1.2 and the new major 1.0.3. 
0.1.3 has been unpublished due to major changes!

**Comment by gchauvet** [comment]
Can you extract a complete sample ?

**Comment by markotom** [comment]
I think all problem is about this lines:
https://github.com/z0mt3c/node-restify-validation/blob/master/lib/error.js#L28-L32

Why does returns `options.errorHandler(res, errors)` if is not a function? Maybe this is a solution:

```js
    if(options.errorHandler && typeof(options.errorHandler) === 'function') {
        return res.send(new options.errorHandler(errors));
    } else {
        return res.send (400, {
            status: 'validation failed',
            errors: errors
        });
    }
```

**Comment by gchauvet** [comment]
Waiting feedback from emmiter on this issue with last merge

**Comment by z0mt3c** [comment]
Go safe an separate stuff...

**Comment by sandijsaploks** [comment]
"node-restify-validation": "^1.0.3",
"restify": "~2.8.3"

Does not work

**Comment by sandijsaploks** [comment]
IF restify error handlers are changing from version to version, what if it is better to create your own errors handler which will stay immutable all the time?

**Comment by z0mt3c** [comment]
released as 1.0.4
